rpc_client/src/client.rs

```diff
@@ -0,0 +1,330 @@
+use std::fmt::{Debug, Formatter, Error as FmtError};
+use std::io::{BufReader, BufRead};
+use std::sync::{Arc, Mutex};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::collections::BTreeMap;
+use std::thread;
+use std::mem;
+use std::time;
+
+use std::path::PathBuf;
+use util::Hashable;
+use url::Url;
+use std::fs::File;
+
+use ws::{connect,
```

Comment:
style: don't import, use `ws::connect` at call sites. _really_ helps code readability. I had to go hunting all over to find where you were getting the magical connect function from.